config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2ClientConfigurerUtils.java

```diff
@@ -96,4 +98,9 @@ private OAuth2ClientConfigurerUtils() {
 		}
 		return (!authorizedClientServiceMap.isEmpty() ? authorizedClientServiceMap.values().iterator().next() : null);
 	}
+
+	static <B extends HttpSecurityBuilder<B>> RestOperations getRestOperationsBean(B builder) {
+		return builder.getSharedObject(ApplicationContext.class).getBean(
```

Comment:
This looks like it would fail if there isn't a bean by that name and type or if there is more than one bean of this type.

Reply:
Is there a reason you think we should use a bean name here? We don't typically do the lookups by bean name.

Reply:
> This looks like it would fail if there isn't a bean by that name and type or if there is more than one bean of this type

`OAuth2ClientRestOperationsPostProcessor` defined in `OAuth2ClientConfiguration` will register the `@Bean` using `OAuth2ClientBeanNames.REST_OPERATIONS` so it won't fail. It's expected to be registered.

> Is there a reason you think we should use a bean name here?

I thought this is what we decided on, as per [comment](https://github.com/spring-projects/spring-security/issues/5607#issuecomment-615496025)

Furthermore, an application may registered one or more `RestOperations` `@Bean` so we need to ensure we pick up the correct `RestOperations` `@Bean` to be used for the client flows.

Also, see tests in `OAuth2ClientConfigurationTests`  starting at `loadContextWhenPostProcessedThenRestOperationsRegistered()` which demonstrates that the `@Bean` is registered by default and how a user could override the `@Bean`

Reply:
Rather than relying on `OAuth2ClientRestOperationsPostProcessor` could we just default the `RestOperations` if none was found? This is what we do in other circumstances.

Reply:
> This is what we do in other circumstances

Can you please be specific and provide a code sample I could reference?

Reply:
Are you referring to the way we conditionally `@Import` using `ImportSelector`? For example, `OAuth2ClientConfigurationOAuth2ClientWebMvcImportSelector`?

Reply:
We often try to get a Bean if it doesn't exist we create a default instance. An [example is PasswordEncoder](https://github.com/spring-projects/spring-security/blob/5.4.0-M1/config/src/main/java/org/springframework/security/config/annotation/authentication/configuration/AuthenticationConfiguration.java#L320-L330)